geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionConfigImpl.java

```diff
@@ -940,23 +941,19 @@ public DistributionConfigImpl(Properties nonDefault, boolean ignoreGemFirePropsF
       attNameSet.add(GEMFIRE_PREFIX + attNames[index]);
     }
 
-    /* clone() is a synchronized method for Properties (actually in Hashtable) */
-    Properties sysProps = (Properties) System.getProperties().clone();
-    Iterator<?> sysPropsIter = sysProps.entrySet().iterator();
-    while (sysPropsIter.hasNext()) {
-      Map.Entry sysEntry = (Map.Entry) sysPropsIter.next();
-      String sysName = (String) sysEntry.getKey();
-      if (attNameSet.contains(sysName) || sysName.startsWith(GEMFIRE_PREFIX + SECURITY_PREFIX_NAME)
-          || sysName.startsWith(GEMFIRE_PREFIX + SSL_SYSTEM_PROPS_NAME)) {
-        String sysValue = (String) sysEntry.getValue();
+    Enumeration<?> sysPropsIter = System.getProperties().propertyNames();
+    while (sysPropsIter.hasMoreElements()) {
```

Comment:
I think this would read better as a `for` loop.
```
    for (Object keyObj : System.getProperties().keySet()) {
      String key = (String) keyObj;
      String sysValue = System.getProperty(key);
      if ((attNameSet.contains(key)
          || key.startsWith(GEMFIRE_PREFIX + SECURITY_PREFIX_NAME)
          || key.startsWith(GEMFIRE_PREFIX + SSL_SYSTEM_PROPS_NAME))
          && sysValue != null) {
        String attName = key.substring(GEMFIRE_PREFIX.length());
        props.put(attName, sysValue);
        this.sourceMap.put(attName, ConfigSource.sysprop());
      }
    }
```